src/facade/dragonfly_listener.cc

```diff
@@ -190,19 +188,24 @@ ProactorBase* Listener::PickConnectionProactor(LinuxSocketBase* sock) {
     total = pp->size();
   }
 
-  if (GetFlag(FLAGS_conn_use_incoming_cpu)) {
+  if (!sock->IsUDS()) {
     int fd = sock->native_handle();
 
     int cpu, napi_id;
     socklen_t len = sizeof(cpu);
 
+    // I suspect that the advantage of using SO_INCOMING_NAPI_ID is that
+    // we can also track the affinity changes during the lifetime of the process
+    // i.e. when a different CPU is assigned to handle the RX traffic.
     CHECK_EQ(0, getsockopt(fd, SOL_SOCKET, SO_INCOMING_CPU, &cpu, &len));
     CHECK_EQ(0, getsockopt(fd, SOL_SOCKET, SO_INCOMING_NAPI_ID, &napi_id, &len));
     VLOG(1) << "CPU/NAPI for connection " << fd << " is " << cpu << "/" << napi_id;
 
-    vector<unsigned> ids = pool()->MapCpuToThreads(cpu);
-    if (!ids.empty()) {
-      id = ids.front();
+    if (GetFlag(FLAGS_conn_use_incoming_cpu)) {
+      vector<unsigned> ids = pool()->MapCpuToThreads(cpu);
+      if (!ids.empty()) {
+        id = ids.front();
+      }
```

Comment:
This could have really just been a reference, but I see its returned by value in helio

Reply:
i could introduce a static empty array

```
vector<unsigned> ProactorPool::MapCpuToThreads(unsigned cpu_id) const {
  if (cpu_id >= cpu_threads_.size()) {
    return vector<unsigned>{};
  }
  return cpu_threads_[cpu_id];
}
```

Reply:
fixed